independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java

```diff
@@ -169,4 +245,36 @@ public int compareTo(InterceptorInfo other) {
         return getTarget().toString().compareTo(other.getTarget().toString());
     }
 
+    private void addCallback(List<MethodInfo> methods, MethodInfo method) {
+        if (!isCallbackOverriden(methods, method)) {
+            methods.add(validateSignature(method));
+        }
+    }
+
+    private boolean isCallbackOverriden(List<MethodInfo> methos, MethodInfo method) {
```

Comment:
Maybe add a comment explaining why this works?

I know that interceptor methods must always have the same parameter list per the specification, and return type doesn't matter, because you can't declare 2 methods that only differ by the return type. However, in ArC, we support 2 different parameter lists: `(InvocationContext)` and `(ArcInvocationContext)`, so you can technically do this:

```java
public class SuperInterceptor {
    @AroundInvoke
    public Object intercept(InvocationContext ctx) { ... } // <1>
}

public class SubInterceptor extends SuperInterceptor {
    @AroundInvoke
    public Object intercept(ArcInvocationContext ctx) { ... } // <2>
}
```

And `<2>` does _not_ override `<1>`. If you try to annotate `<2>` with `@Override`, the code won't compile.

I know it's a corner case.

Reply:
On the other hand, we may want to treat this case _as if_ `<2>` did override `<1>`... :shrug:

Reply:
> On the other hand, we may want to treat this case _as if_ `<2>` did override `<1>`... shrug

I think that we must treat this case this way...

> Maybe add a comment explaining why this works?

:+1:

Reply:
In fact, I need to modify the logic a little bit because the spec wording is: "If an interceptor method is overridden by another method (regardless whether that method is itself an interceptor method), it will not be invoked." and we don't comply with the "regardless" part...

Reply:
> > On the other hand, we may want to treat this case _as if_ `<2>` did override `<1>`... shrug
> 
> I think that we must treat this case this way...

FTR, I think we do the exact same thing with Weld which also has its custom IC variant.